### PR TITLE
Test to assert guest user has permission denied to create or upload files

### PR DESCRIPTION
**Pull request by jasson99**
## Description
Add acceptance tests to assert guest user has permission denied to create or upload files

## Related Issue

- Fixes https://github.com/owncloud/core/issues/36712

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [ ] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have read the **CONTRIBUTING** document.
- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.

### CODE REVIEW

File: tests/acceptance/features/webUIGuests/guests.feature [hunk 1]
```diff
@@ -128,4 +128,16 @@ Feature: Guests
...
-      | user@example.com | USER@example.com | user@EXAMPLE.com |
\ No newline at end of file
+      | user@example.com | USER@example.com | user@EXAMPLE.com |
+
+    @mailhog
+    Scenario: Guest user is not able to upload or create files
+      Given user "user0" has been created with default attributes and skeleton files
+      And the administrator has created guest user "guest" with email "guest@example.com"
+      And user "user0" has created folder "/tmp"
+      And user "user0" has shared folder "/tmp" with user "guest@example.com"
```
Comment (HariBhandari07) [inline tests/acceptance/features/webUIGuests/guests.feature]: both administrator and sharing action is not required to create guest user

Comment (jasson99) [reply tests/acceptance/features/webUIGuests/guests.feature]: The steps for sharing with the guest user has been done using webUI only. But for this scenario, we need that sharing step to be performed in the given step and it needs to be done using api..So, I thought this scenario would be okay as it is right now, for now.

File: tests/acceptance/features/webUIGuests/guests.feature [hunk 2]
```diff
@@ -128,4 +128,16 @@ Feature: Guests
...
+      When guest user "guest" registers and sets password to "password" using the webUI
+      And user "guest@example.com" logs in using the webUI
+      Then the user should be redirected to a webUI page with the title "Files - %productname%"
+      And the user should not have permission to upload or create files
```
[the remark this answers is not in the document]
Comment (jasson99) [reply tests/acceptance/features/webUIGuests/guests.feature]: the plural for permission is also permission

File: tests/acceptance/features/webUIGuests/guests.feature [hunk 3]
```diff
@@ -128,4 +128,13 @@ Feature: Guests
       | USER@example.com | user@example.com | user@example.com |
       | USER@example.com | USER@example.com | USER@example.com |
       | USER@example.com | USER@example.com | user@example.com |
-      | user@example.com | USER@example.com | user@EXAMPLE.com |
\ No newline at end of file
+      | user@example.com | USER@example.com | user@EXAMPLE.com |
+
+    @mailhog
+    Scenario: Guest user is not able to upload or create files
```
Comment (phil-davis) [inline tests/acceptance/features/webUIGuests/guests.feature]: We could also add a scenario where `user0` shares a folder (with read-only permissions), then the guest logs in and opens the folder. `Then the user should not have permission to upload or create files`

**Pull request merged**